readers/mutation_reader.cc

```diff
@@ -226,6 +226,9 @@ bool mutation_fragment_stream_validating_filter::operator()(mutation_fragment_v2
         } else if (_validation_level >= mutation_fragment_stream_validation_level::partition_key) {
             on_validation_error(mrlog, format("[validator {} for {}] Unexpected mutation fragment: partition key {}: previous {}, current {}",
                     static_cast<void*>(this), _name, _validator.previous_partition_key(), _validator.previous_mutation_fragment_kind(), kind));
+        } else if (kind == mutation_fragment_v2::kind::partition_end && _validator.current_tombstone()) {
```

Comment:
Unrelated to this patch, but do we try to recreate the reason here, rather than have the _validator return the reason for validation error? Seems like this is should be the responsibility of the low level validator.

Reply:
Good point. I find myself composing the same messages over and over again. It would be better if the low-level validator did this.

Reply:
I will do this in a follow-up.